### ISSUE TITLE
Players who login later, recieve less knockback

### ISSUE THREAD

**Issue by Odie680** [PaperSpigot/Paper#43]
Like the title says, if a player logins later or the last, they take little to no knockback compared to other players.

I've searched it up, and it seems to me that SportBukkit has fixed the issue.

https://github.com/OvercastNetwork/SportBukkit/commit/0a6819e11487d6fef0b5bdc4fbd635ef8a512101

Thank you :)

**Comment by Odie680** [comment]
Oh and this is my versoin:

https://i.bmc.la/431188230813856.png

**Comment by gsand** [comment]
Set your player-shuffle to 200 in the spigot.yml file.

**Comment by Odie680** [comment]
That's not really a fix. There is always 1 person with reduced knockback.

**Comment by gsand** [comment]
Anything is better than nothing.

**Comment by Odie680** [comment]
Yes, I understand but giving a player 200 ticks of antikb on a player versus player server means you can easily get a 10 second combo. It messes the gameplay up.

This is not a solution/fix.

**Comment by Ichbinjoe** [comment]
So commit 9383504e52d8a05ca6f9b5b67b1b7afbcc18998d initially fixed this issue, with commits 3f758a20de02dcd9c55e82f2a01d82ebfcbb799e and cebd81ef1002b51c14d79f2a7ce11f2a38c70683 also dealt with this, however commit b56ce6c1be2c92a0fee2030e7524e0ac3a0f63b0 reverted all of these changes, stating that it needs to be dealt with in a better way, and made more configurable. I made my commit before I read that, and that's why I had created then closed the pull request. I guess my question is how would you see it being done better/being able to be configured better?

**Comment by Zbob750** [comment]
@Ichbinjoe That's a good question, and one we're open to input on. The issue with sportbukkit's fix was that it sent knockback in it's most "raw" form, before anything else had any effect on the velocity. Which made it very extreme all of the time. If nothing else, we could make it configurable and perhaps apply a bit of randomness to make it seem a tad more realistic?

Again, we're open to input on this :)

**Comment by Ichbinjoe** [comment]
@Zbob750 Well, wasn't the issue that the kb velocity was actually being cancelled out once the player's movement is calculated later in the tick? So in theory, shouldn't the kb be that power all the time? I'm not sure of other things that would effect how the kb is calculated. The way Sportbukkit fixes the issue I feel is completely legitimate. Sure we can add a value to try to scale down this kb value, but I think that should be handled in a different pull request entirely (more of a feature addition, not a bug fix). In addition, I think we should have a boolean on off on whether we actually want this calculation to occur: how the current code is written at the moment is how minecraft operates in vanilla, while this patch would be modifying vanilla behavior. I think the randomness is a design decision by the project. I'm not sure how well it would be received if we did a system like this. In interest of operating in vanilla, I don't think we should add a random kb modifier. If a server wants that, they can fork the project and add it. I think the original patch handled kb correctly. That's just my opinion on the matter, so yeah lol. Anyone else including people who use this in production environments should chime in. :P

**Comment by Zbob750** [comment]
We'd make the entire fix configurable either way. The main issue with SB's fix was how extreme it was, we removed it to decide if there was a better way to handle it. I'd be fine with using their implementation with a modifier.

The main intention I had behind the "randomness" was to make it seem semi-realistic and less... robotic (not sure if that's the word I'm looking for) . That being said, I do not do a ton of MC PvP gameplay and I recognize that it means I can be out of touch with PvP style server owners. We usually run any PvP gameplay options through the people in IRC before pushing anything (and obviously here would be another fine place to discuss any of that).

Still entirely open to everyone's thoughts on this. As you said, we definitely need more feedback from the community before making any major decisions on this issue.
